main.go

```diff
@@ -0,0 +1,106 @@
+package main
```

Comment:
This package seems to have a lot of responsibilities now. Seems like connection setup is one big concern and api/server setup is another distict concern. Can they be separated?

Reply:
i could see that. maybe open a ticket though; doesn't seem like worth blocking this over.